src/gennylib/include/gennylib/service.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021-present MongoDB Inc.
```

Comment:
I added this file because it made unit testing `Topology` easier. Another option would be to templatize it but I wanted to avoid having all the details of discovery in the header file.

Reply:
Note to self to come back here.

I think this is a good design. Genny doesn't use a lot of virtuals in its core code but virtuals are a great strategy to ease testing in utilities.

But `service` is a bit of a smurf term. Perhaps it makes sense to rename these and make them have a more concrete connection (both in name and code location) to `topology.hpp`.

Reply:
I renamed it to `Connection`. Part of why I put it in a separate file is I thought it might be useful elsewhere in other places, the only other way we have to test driver-level things is with the resmoke integration and APM stuff, which might be more involved. Let me know though if we should just cross that bridge when we get there and I'll put it in the `topology.hpp` file.